### PR TITLE
Make some of the Flags.cpp stuff simpler

### PR DESCRIPTION
**Pull request by GabrielRavier**
This changes some of the code in Flags.cpp to macros. This simplifies the code (without changing what is actually compiled on MSVC) and most likely makes it more accurate (it would seem really unlikely (at least to me) that Pixel actually copied the same code thrice)

### CODE REVIEW

**Comment by Clownacy** [comment]
I prefer my #defines to come after the #includes, and calling them SET_FLAG/UNSET_FLAG/etc. instead of SET_BIT makes more sense to me. Also, aren't the semicolons at the end of the lines that use SET_BIT/UNSET_BIT useless?

**Comment by GabrielRavier** [comment]
> I prefer my #defines to come after the #includes

I don't see any defines before includes.
> Aren't the semicolons at the end of the lines that use SET_BIT/UNSET_BIT useless?

I think I'm just going to remove them from the macros instead of in the lines that use the macros (it feels ugly not to have a semicolon at eol)
> I prefer [...] calling them SET_FLAG/UNSET_FLAG/etc.

I'd say calling them SET_BIT/UNSET_BIT makes it much more clear what the code does

**Comment by Clownacy** [comment]
> I don't see any defines before includes.
I mean _right_ after the #includes.

> I think calling them SET_BIT/UNSET_BIT makes it much more clear what the code does
There's not really anything I can say besides 'I disagree'.

**Comment by GabrielRavier** [comment]
> I mean right after the #includes.

Ok I'll try putting them as far as possible from the includes

> There's not really anything I can say besides 'I disagree'.
:frowning:. I'll change the names and make longer documentation I guess

**Comment by Clownacy** [comment]
No, I mean I prefer them _directly after_ the includes, so all the preprocessor commands are together.

**Comment by GabrielRavier** [comment]
Oh ok doing that

**Comment by GabrielRavier** [comment]
Fixed that

**Comment by Clownacy** [comment]
I'd rather not have redundant macros.

**Comment by GabrielRavier** [comment]
Okay I'll remove them

**Comment by GabrielRavier** [comment]
I removed them

File: src/Flags.cpp [hunk 1]
```diff
@@ -4,6 +4,12 @@
 
 #include "WindowsWrapper.h"
 
+// Macros for setting, un-setting and getting bits
+// Each flag is stored in a bit, so we can use the exact same macros for this (not defining BIT macros for conciseness)
```
Comment (Clownacy) [inline src/Flags.cpp]: This line feels unnecessary.

Comment (GabrielRavier) [reply src/Flags.cpp]: I tried simplifying it a bit

**Pull request merged**